### ISSUE TITLE
Document or remove dependence of RegExRemove Processor on outputs

### ISSUE THREAD

**Issue by charlesfrye** [jupyter/nbconvert#1091]
The `RegExRemoveProcessor` removes cells if they match the given regular expression and have no output.

This is not made clear in the docstring, but feels like surprising behavior to me. I often want to remove cells based only on their contents, not on whether they have outputs, and in fact often want to remove cells that produce outputs, e.g. solution cells in an educational notebook. As it stands, without looking at the source, there is no way to know that the outputs of the cell influence this preprocessor, which makes debugging difficult. My current work-around is to use the `ClearOutputPreprocessor` first, but that won't work if I need to preserve some, but not all, outputs.

If that is expected and desired behavior, it should be noted [here, in the docstring](https://github.com/jupyter/nbconvert/blob/2e0baa25160ebf342f798198a28ee5364ebcd4fa/nbconvert/preprocessors/regexremove.py#L20), perhaps by changing the linked line to:

```
    of unicode strings. If the contents match any of the patterns and the cell
    has no outputs, the cell is removed from the notebook.
```

But I would personally prefer the behavior in [this version](https://github.com/jupyter/nbconvert/blame/3860a848ccfde8e86f5b278fe85177f8786c663c/nbconvert/preprocessors/regexremove.py), which simply removes cells that match, or a flag to switch between the two behaviors.

**Nbconvert version:** Any after a8cb3a5, linked above (.

Thanks for making a box of really nice tools! Couldn't effectively teach without them.

**Comment by MSeal** [comment]
Given that, @charlesfrye I'm for changing the behavior to what was documented. It makes a lot more sense as an expected behavior imo. Do you want to open a PR for the change?

**Comment by charlesfrye** [comment]
@MSeal , which branch should I make the PR onto?

It was a quick change, so if the conversation on the PR goes the other way, I can instead just add documentation.

**Comment by MSeal** [comment]
Making the PR against `master` is fine. There's a `6.x` branch but that's the only other active merge location atm,

**Comment by takluyver** [comment]
Closing as it looks like this was fixed by #1095 - please reopen or let me know if I'm wrong.
